lldb/test/Shell/SwiftREPL/ExistentialTypes.test

```diff
@@ -0,0 +1,41 @@
+
+// Test that the we can resolve the dynamic type of existential correctly.
+// REQUIRES: swift
+
+// RUN: %lldb --repl < %s | FileCheck %s
+
+protocol A {}
+
+class B: A {}
+
+struct C: A {}
+
+enum D: A {
+  case empty
+}
+
+class E: Error, A {}
+
+struct F: Error, A {}
+
+enum G: Error, A { 
```

Comment:
Is there something special about `Error`?

Reply:
In SwiftASTContext the Error type is special-cased in a couple of places. I think in TypeSystemSwiftTypeRef it's mostly just another protocol.

Reply:
https://github.com/apple/llvm-project/blob/3c96a2f1c42ab1cf57019e903b03fb1eac682114/lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp#L6629

Reply:
https://github.com/apple/llvm-project/blob/3c96a2f1c42ab1cf57019e903b03fb1eac682114/lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp#L5387

Reply:
When we try to find the dynamic type of some existential, errors are one of the possible existential cases: https://github.com/apple/swift/pull/37673/files#diff-22f5e159ab66c32ef3537a5a7c2617427545535dd1e7899a837cbf9effe69be1R936